backend/layers/thirdparty/schema_validator_provider.py

```diff
@@ -109,12 +109,9 @@ def check_anndata_requires_fragment(self, anndata_file) -> bool:
 
         return atac_seq.check_anndata_requires_fragment(anndata_file)
 
-    def deduplicate_fragments(self, fragment_file: str) -> None:
-        import os
-
+    def deduplicate_fragments(self, fragment_file: str) -> str:
         import cellxgene_schema.atac_seq as atac_seq
 
         output_file = fragment_file + ".deduplicated"
         atac_seq.deduplicate_fragment_rows(fragment_file, output_file)
-        os.remove(fragment_file)
-        os.rename(output_file, fragment_file)
+        return output_file
```

Comment:
Can you explain this change? I'm not seeing the reason for it from the PR description